src/topology/metric_space/basic.lean

```diff
@@ -1,23 +1,31 @@
 /-
 Copyright (c) 2015, 2017 Jeremy Avigad. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Metric spaces.
-
 Authors: Jeremy Avigad, Robert Y. Lewis, Johannes Hölzl, Mario Carneiro, Sébastien Gouëzel
+-/
+
+import topology.metric_space.emetric_space
+import topology.shrinking_lemma
+import topology.algebra.ordered
+import data.fintype.intervals
 
-Many definitions and theorems expected on metric spaces are already introduced on uniform spaces and
-topological spaces. For example:
-  open and closed sets, compactness, completeness, continuity and uniform continuity
+/-!
+# Metric spaces
+
+This file defines metric spaces. Many definitions and theorems expected
+on metric spaces are already introduced on uniform spaces and topological spaces.
+For example: open and closed sets, compactness, completeness, continuity and uniform continuity
+
```

Comment:
It'd be good to include a section listing the main declarations in the file, cf. https://leanprover-community.github.io/mathlib_docs/geometry/manifold/charted_space.html . Note that fully-qualified names in backticks get turned into links in the generated HTML.

Reply:
Alright, I'll try do this! Obviously won't be as good as if they were written at the time by the authors, but I'll have a scroll through and make something passable :)

Reply:
Thanks for doing this. Bryan's comment also applies for the other file. Also, it is perfectly fine to have only 2-3 of these in one PR (even one if it is a larger file). This makes it easier to review.

Reply:
So whilst just having a cursory look at the files in `style_exceptions.txt`, I noticed a _lot_ of them have just raw errors that annoy the linter, from either putting `universes u` or `open my_stuff` above the docstring (which thankfully the docstring generator can ignore) to forgetting to change `/-` to `/-!` (which the parser sadly can't ignore, c.f. [here](https://leanprover-community.github.io/mathlib_docs/data/matrix/pequiv.html#pequiv) vs [its source](https://github.com/leanprover-community/mathlib/blob/master/src/data/matrix/pequiv.lean). I'm rededicating this PR to fixing these, and then I'll start on fixing some partial documentation into more actual documentation later.

Reply:
Yep, that sounds reasonable.